src/storage/engines/SupabaseStorageEngine.ts

```diff
@@ -347,7 +347,13 @@ export class SupabaseStorageEngine extends CloudStorageEngine {
       // get the metadata field from the data object
       const metadata = data[0].data;
       if (metadata) {
-        return metadata;
+        const cleanedModes = cleanupModes(metadata as Record<string, boolean>);
+        await this.supabase
+          .from('revisit')
+          .update({ data: cleanedModes })
+          .eq('studyId', `${this.collectionPrefix}${studyId}`)
+          .eq('docId', 'metadata');
+        return cleanedModes;
```

Comment:
This is guaranteed to send a request every time we pull the modes. We should design this in a way to minimize calls to the DB. That is, we should add a check here and only do the cleanup if necessary